applications/build-modular-monolith-with-platformatic/services/media-service/plugin.js

```diff
@@ -0,0 +1,52 @@
+/// <reference path="../books-service/clients/persons-service/persons_service.d.ts" />
+// TODO: The client will be moved if I can get things working in the Composer `onRoute` hook.
+
+'use strict'
+
+/** @param {import('fastify').FastifyInstance} app */
+module.exports = async function (app) {
+
+  // /books/
+  // /books/{id}
+
+  app.log.info('Media service plugin loaded')
+
+  // Modify route: GET /books
+  app.addComposerOnRouteHook('/books/', ['GET'], routeOptions => {
+    const responseSchema = routeOptions.schema.response[200].items
+    responseSchema.properties.authorName = {
+      type: 'string'
+    }
+    responseSchema.required.push('authorName')
+
+    // From the Fastify docs for the `preSerialization` hook:
+    //
+    // > Note: the hook is NOT called if the payload is a string, a Buffer, a stream, or null.
+    //
+    // https://www.fastify.io/docs/latest/Reference/Hooks/#preserialization
+    //
+    routeOptions.preSerialization = async (request, reply, payload) => {
+      // This message is never logged
+      app.log.info('CALLED: preSerialization')
+
+      return {
+        ...payload,
+        // TODO: Retrieve the author name from the Persons service using a client
+        authorName: 'Some Author'
+      }
+    }
```

Comment:
This isn't working. The `preSerialization` hook isn’t called for the `GET /books/` route as the payload is a stream.

From the Fastify docs for the `preSerialization` hook:

> Note: the hook is NOT called if the payload is a string, a Buffer, a stream, or null.

https://www.fastify.io/docs/latest/Reference/Hooks/#preserialization

The Platformatic [Composer API modification](https://oss.platformatic.dev/docs/reference/composer/api-modification/) docs give an example that sets a `preSerialization` hook, but it's possible that example doesn't work as I don't see test coverage for this use case ([feature PR](https://github.com/platformatic/platformatic/pull/1067/files)).

Reply:
@ivan-tymoshenko PTAL